### PR TITLE
vo_conesearch API changes

### LINKED ISSUE #1048: ENH: ConeSearch to accept keyword to set URL
As discussed in https://github.com/adamginsburg/astroquery-paper/pull/31#issuecomment-367045801 .

I have to think about whether to have the keyword set a single URL, or mimic a more complicated `catalog_db` option in the "classic" `conesearch`.

### LINKED ISSUE #1522: vo_conesearch should return astropy Table rather than  votable

### LINKED ISSUE #1523: vo_conesearch to more informative `"search returned no results.`
Currently the traceback is a bit misleading, as not getting a result is totally expected (the survey doesn't cover that part of the sky). Some of the other modules are returning a NoResultWarning in these situations that is much nicer. 

```
>>> from astroquery.vo_conesearch import conesearch
>>> from astropy.table import Table
>>> result = conesearch.conesearch((120, -40), radius=0.1, catalog_db="http://localhost:8080/cone?", timeout=4*60)
WARNING: W25: ?:?:?: W25: 'http://localhost:8080/cone?' failed with: Catalog server 'http://localhost:8080/cone?' returned 0 result [astroquery.vo_conesearch.conesearch]

---------------------------------------------------------------------------
VOSError                                  Traceback (most recent call last)
<ipython-input-3-4845e9eaf26f> in <module>()
      1 from astroquery.vo_conesearch import conesearch
      2 from astropy.table import Table
----> 3 result = conesearch.conesearch((120, -40), radius=0.1, catalog_db="http://localhost:8080/cone?", timeout=4*60)
      4 result

/epyc/opt/anaconda/lib/python3.6/site-packages/astroquery/vo_conesearch/conesearch.py in conesearch(center, radius, verb, catalog_db, pedantic, verbose, cache, timeout, query_all)
    249         if n_timed_out > 0:
    250             err_msg += ' ({0} URL(s) timed out.)'.format(n_timed_out)
--> 251         raise VOSError(err_msg)
    252 
    253     return result

VOSError: None of the available catalogs returned valid results.
```

cc @pllim

### PR DESCRIPTION
**Pull request by pllim**
**Please milestone this only for a major release: Contains non-backward compatible API changes!**

**TODO**

- [x] Add change logs.

Fix #1523
- [x] Code implementation
- [x] Fixed tests
- [x] Documentation update

Fix #1522 
- [x] Code implementation
- [x] Fixed tests
- [x] Documentation update

Fix #1048 (also see https://github.com/adamginsburg/astroquery-paper/pull/31#issuecomment-367045801)
- [x] Code implementation
- [x] Fixed tests
- [x] Documentation update

cc @bsipocz and @keflavich

### CODE REVIEW

**Comment by pllim** [comment]
CI failed with this -- Is this known to be transient or is this something else? 🤔 
```
Collecting cdshealpix (from mocpy>=0.5.2)
  ERROR: Could not find a version that satisfies the requirement cdshealpix (from mocpy>=0.5.2) (from versions: none)
ERROR: No matching distribution found for cdshealpix (from mocpy>=0.5.2)
```

**Comment by pllim** [comment]
@bsipocz , this is now ready for review. Thanks!

**Comment by bsipocz** [comment]
@pllim - not sure what is a major release in terms of astroquery. I'm looking around now for PRs ready to be merged before tagging 0.3.10 in the coming week. Would it be a major release, or shall we wait for the next one? However after 0.3.10 is out, we'll switch back to CD, so I suppose it might be better to merge it for a manually tagged and conda packaged one rather than a normal CD one.

What do you think? Does this break many tutorials, etc downstream?

File: astroquery/vo_conesearch/conesearch.py [hunk 1]
```diff
@@ -78,21 +74,33 @@ class AsyncConeSearch(AsyncBase):
...
+def conesearch(center, radius, verb=1, catalog_db=None,
+               verbose=True, cache=True, query_all=False,
+               return_astropy_table=True, use_names_over_ids=False):
     """
     Perform Cone Search and returns the result of the
     first successful query.
 
+    .. note::
+
+        Use ``astroquery.vo_conesearch.conf.pedantic`` to control
```
Comment (bsipocz) [inline astroquery/vo_conesearch/conesearch.py]: should we rather deprecate this first? you know better the userbase to judge it.

File: astroquery/vo_conesearch/conesearch.py [hunk 2]
```diff
@@ -223,32 +220,31 @@ def conesearch(center, radius, verb=1, catalog_db=None, pedantic=None,
...
-        cs.URL = url
-
         if verbose:  # pragma: no cover
             color_print('Trying {0}'.format(url), 'green')
 
         try:
-            r = cs.query_region(center, radius, verb=verb, cache=cache,
-                                verbose=verbose)
-
+            r = ConeSearch.query_region(
```
Comment (bsipocz) [inline astroquery/vo_conesearch/conesearch.py]: can be done in follow-up, but would be nice to get rid of the single letter variables and rather just call this `result` or `response` or similar.

File: astroquery/vo_conesearch/conesearch.py (hunk 1 above)
Comment (pllim) [reply astroquery/vo_conesearch/conesearch.py]: User base is not wide enough to justify the maintenance overhead.

File: astroquery/vo_conesearch/conesearch.py (hunk 2 above)
Comment (pllim) [reply astroquery/vo_conesearch/conesearch.py]: :sweat_smile:  It is a valid point. I'll do it in a follow-up commit when a full review is given.

**Comment by pllim** [comment]
> Does this break many tutorials, etc downstream?

Yes, it does. Let's wait for 0.4.

File: astroquery/vo_conesearch/conesearch.py (hunk 2 above)
Comment (pllim) [reply astroquery/vo_conesearch/conesearch.py]: Thinking about this more, I now prefer to revisit renaming short variables as a separate PR, just to make my life saner. 😬

**Comment by pllim** [comment]
@bsipocz , should I just rebase manually? I am also happy to let you do it with the bot if you want.

**Comment by bsipocz** [comment]
Just rebase the usual way, I don't have time at all to play with the bot/other actions.

**Comment by pllim** [comment]
Here's a local test run with remote data after the rebase:
```
python setup.py test -P vo_conesearch --remote-data
```
```
platform linux -- Python 3.7.3, pytest-5.0.1, py-1.8.0, pluggy-0.12.0

Running tests with astroquery version 0.4.dev5751_testrun.
Running tests in astroquery/vo_conesearch docs/vo_conesearch.

Date: 2019-10-09T21:19:09

Platform: Linux-3.10.0-1062.1.2.el7.x86_64-x86_64-with-redhat-7.7-Maipo

Executable: .../miniconda/envs/py37/bin/python

Full Python Version: 
3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 21:52:21) 
[GCC 7.3.0]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Numpy: 1.17.0
Matplotlib: 3.1.1
Astropy: 4.0.dev25805
APLpy: not available
pyregion: not available
pyVO: not available
astropy_helpers: 1.3
Using Astropy options: remote_data: any.

rootdir: ..., inifile: setup.cfg
plugins: doctestplus-0.4.0, remotedata-0.3.1, openfiles-0.3.2, arraydiff-0.3, ci-watson-0.0.0.dev76+g4b73d3b5, asdf-2.4.0.dev63+gb67cb02
collected 51 items                                                                                 

astroquery/vo_conesearch/tests/test_conesearch.py ...................x....                   [ 47%]
astroquery/vo_conesearch/tests/test_vos_catalog.py ...............                           [ 76%]
astroquery/vo_conesearch/validator/tests/test_inpect.py ....                                 [ 84%]
astroquery/vo_conesearch/validator/tests/test_validate.py .....                              [ 94%]
docs/vo_conesearch/client.rst .                                                              [ 96%]
docs/vo_conesearch/validator.rst .                                                           [ 98%]
docs/vo_conesearch/vo_conesearch.rst .                                                       [100%]

=== 50 passed, 1 xfailed in 25.04 seconds ===
```

File: astroquery/vo_conesearch/conesearch.py [hunk 3]
```diff
@@ -1,10 +1,5 @@
...
-"""
-Support VO Simple Cone Search capabilities.
-
-.. note:: This maintains a similar API as ``astropy.vo.client``.
-
-"""
+"""Support VO Simple Cone Search capabilities."""
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 import six
```
Comment (bsipocz) [inline astroquery/vo_conesearch/conesearch.py]: Clean up time if you want. But actually it's reasonable to leave that as future fun :)

File: astroquery/vo_conesearch/conesearch.py [hunk 4]
```diff
@@ -78,21 +74,33 @@ class AsyncConeSearch(AsyncBase):
...
+    74271
 
     """
     def __init__(self, *args, **kwargs):
         super(AsyncConeSearch, self).__init__(conesearch, *args, **kwargs)
 
 
-def conesearch(center, radius, verb=1, catalog_db=None, pedantic=None,
-               verbose=True, cache=True, timeout=None, query_all=False):
+def conesearch(center, radius, verb=1, catalog_db=None,
```
Comment (bsipocz) [inline astroquery/vo_conesearch/conesearch.py]: I wonder whether we can take these kwargs as keyword only arguments, or that would upset downstream too much? At least that would make sure that the removal of `pedantic` etc doesn't hide in their code and cause cryptic issues.

File: astroquery/vo_conesearch/conesearch.py [hunk 5]
```diff
@@ -172,31 +173,32 @@ def conesearch(center, radius, verb=1, catalog_db=None, pedantic=None,
...
+        are not guaranteed to be unique, this may cause some columns
+        to be renamed by appending numbers to the end.  Otherwise
+        (default), use the ID attributes as the column names.
+
     Returns
     -------
-    obj : `astropy.io.votable.tree.Table`
+    obj : `astropy.table.Table` or `astropy.io.votable.tree.Table`
         First table from first successful VO service request.
+        See ``return_astropy_table`` option for the kind of table returned.
```
Comment (bsipocz) [inline astroquery/vo_conesearch/conesearch.py]:
```suggestion
        See ``return_astropy_table`` parameter for the kind of table returned.
```

File: astroquery/vo_conesearch/conesearch.py (hunk 2 above)
Comment (bsipocz) [reply astroquery/vo_conesearch/conesearch.py]: OK, opened: https://github.com/astropy/astroquery/issues/1572 for it.

**Comment by bsipocz** [comment]
I've given this another skimming through and nothing obviously wrong pops up. 

Given that most of these things came up when discussing with @ctslater it would be very nice if he could give it a try before we deploy it to the wild.

File: astroquery/vo_conesearch/conesearch.py (hunk 3 above)
Comment (pllim) [reply astroquery/vo_conesearch/conesearch.py]: Done! 😄

File: astroquery/vo_conesearch/conesearch.py (hunk 4 above)
Comment (pllim) [reply astroquery/vo_conesearch/conesearch.py]: Yeah, fine by me. Done.

File: astroquery/vo_conesearch/conesearch.py [hunk 6]
```diff
@@ -78,21 +71,33 @@ class AsyncConeSearch(AsyncBase):
...
+    74271
 
     """
     def __init__(self, *args, **kwargs):
         super(AsyncConeSearch, self).__init__(conesearch, *args, **kwargs)
 
 
-def conesearch(center, radius, verb=1, catalog_db=None, pedantic=None,
-               verbose=True, cache=True, timeout=None, query_all=False):
+def conesearch(center, radius, *, verb=1, catalog_db=None,
```
Comment (pllim) [inline astroquery/vo_conesearch/conesearch.py]: @bsipocz , this is the right way to do it, right?

p.s. I reran the remote tests and they still passed.

**Comment by bsipocz** [comment]
Thanks @pllim. I would suggest to keep this open to get some testing first.

**Comment by keflavich** [comment]
This passed tests on Oct 10 and has been sitting here.  Should we merge it now?

**Comment by bsipocz** [comment]
@keflavich - I hoped that we can get some testing of this before merging.

@ctslater - would you mind to refresh the ZTF conesearch prototype you did a few month ago trying out the changes from here?

**Comment by ctslater** [comment]
I tried this out with some of the queries and edge cases that had given me trouble before, and the results all look much better this time. Really nice to get Astropy Tables returned straight away, and the zero-results handling is much friendlier.

**Comment by bsipocz** [comment]
Thank you @ctslater for the testing and feedback!

**Comment by bsipocz** [comment]
Thanks @pllim!

**Pull request merged**